src/main/java/seedu/address/logic/commands/people/PeopleReceivedCommand.java

```diff
@@ -0,0 +1,134 @@
+package seedu.address.logic.commands.people;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliPrefix.PEOPLE_COMMAND_TYPE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TRANSACTION_INDEX;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Debt;
+import seedu.address.model.transaction.Loan;
+import seedu.address.model.transaction.TransactionList;
+
+/**
+ * Records that the user has received the amount money lent to a person,
+ * specified by an index in the address book.
+ */
+public class PeopleReceivedCommand extends Command {
+
+    public static final String COMMAND_WORD = "received";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Records that the person has paid you back.\n"
+            + "Parameters: <person's index> (must be a positive integer) [" + PREFIX_TRANSACTION_INDEX
+            + "<loans's index> (must be a positive integer)]\n"
+            + "Example: " + PEOPLE_COMMAND_TYPE + " "
+            + COMMAND_WORD + " 4 i/1";
+
+    public static final String MESSAGE_RECEIVED_SUCCESS = "Removed loan to %1$s by %2$s. %1$s now owes you %3$s.";
+    public static final String MESSAGE_NO_LOAN = "%1$s does not owe you money :(";
+
+    private final Index targetPersonIndex;
+    private final Index targetLoanIndex;
+
+    public PeopleReceivedCommand(Index targetIndex, Index targetLoanIndex) {
+        this.targetPersonIndex = targetIndex;
+        this.targetLoanIndex = targetLoanIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
This function abit long so maybe good to extract out code e.g. from within the if-else loops